QAChat/Data_Processing/pdf_reader.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: 2023 Emanuel Erben
+# SPDX-FileCopyrightText: 2023 Felix Nützel
+
+
+import io
+from datetime import datetime
+
+from pdfminer.converter import TextConverter
+from pdfminer.layout import LAParams
+from pdfminer.pdfdocument import PDFDocument
+from pdfminer.pdfinterp import PDFResourceManager, PDFPageInterpreter
+from pdfminer.pdfpage import PDFPage
+from pdfminer.pdfparser import PDFParser
+from pdf2image import convert_from_bytes
+import pytesseract
+import os
+import nltk
+from QAChat.Data_Processing.document_embedder import DataInformation, DataSource
+
+TESSDATA_PREFIX = os.getenv('TESSDATA_PREFIX')
+nltk.download('punkt')
+
+
+def read_pdf(pdf_object, before: datetime, after: datetime, data_src: DataSource):
+    # First try to read the PDF with PDFMiner
+    text = __read_pdf_from_object(pdf_object)
+
+    # If the result is empty or too short, try OCR
+    if len(text.strip()) < 100:
```

Comment:
With the examples in the main I don't jump into this case

Reply:
Yeah this case is for when we tested it with a fotocopy of a document transformed into a pdf. As the read_pdf_from_object in this case could not read any of the text. To then still get a as good result we added a way to also read text from images. So the pdf gets transformed into an image and then we can read the data. This enables us to also read everything from the text, as well as it is possible

Reply:
The < 100 is son we only do this when we couldn't get any text from the document. Especially large documents are very resource intensive when analyzed by the second function